amads/pitch/pitch_mean.py

```diff
@@ -10,8 +10,13 @@ def pitch_mean(score, weighted=False):
     """
     sum = 0
     count = 0
-    for note in score.find_all(Note):
-        w = note.duration if weighted else 1
-        sum += note.keynum * w
-        count += w
+    if weighted:  # no need to merge tied notes
+        for note in score.find_all(Note):
+            sum += note.keynum * note.duration
+            count += note.duration
+    else:
+        score = score.merge_tied_notes()
```

Comment:
I suppose it would be a bit more efficient to skip the `merge_tied_notes` step and instead when iterating over notes just skip notes where `note.tie in ["continue", "stop"]`?

Reply:
Yes - good idea. I also thought about trying to represent tied notes with some different classes so that score.find_all(Note) would retrieve only the first note of tied note groups and somehow, the object would store a representation of the entire tied group, but that could be confusing. Alternatively, tied notes could actually form linked list where note.tie is a reference directly to the tied note. Probably easy to do in the score construction. Maybe duration and offset would refer to the entire group and additional properties would allow access to the components of the tied group.